server.js

```diff
@@ -1066,6 +1066,81 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Liberapay integration.
+camp.route(/^\/liberapay\/(.*)\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var entity = match[1];  // eg, 'Changaco'
+  var type = match[2];
+  var format = match[3];
+  var apiUrl = 'https://liberapay.com/' + entity + '/public.json';
+  // Establish default
+  var badgeData = getBadgeData('receives', data);
+  switch(type) {
+      case 'gives':
+          badgeData = getBadgeData('gives', data);
+          break;
+      case 'patrons':
+          badgeData = getBadgeData('patrons', data);
+          break;
+      case 'goal':
+          badgeData = getBadgeData('goal progress', data);
+          break;
+      }
+  if (badgeData.template === 'social') {
+    badgeData.logo = getLogo('liberapay', data);
+  }
+  request(apiUrl, function dealWithData(err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      var data = JSON.parse(buffer);
+      var value;
+      var currency;
+      switch(type) {
+        case 'receives':
+            if (data.receiving) {
+                value = data.receiving.amount;
+                currency = data.receiving.currency;
+                badgeData.text[1] = metric(value) + ' ' + currency + '/week';
+                }
+            break;
+        case 'gives':
+            if (data.giving) {
+                value = data.giving.amount;
+                currency = data.giving.currency;
+                badgeData.text[1] = metric(value) + ' ' + currency + '/week';
+                }
```

Comment:
I'm not sure I understand the `else` case. Why does a missing `data.giving` mean anonymous?

Reply:
`data.giving`, `data.receiving`, and `data.goal` each end up as `null` in the JSON when they are hidden by the user at Liberapay.com.

Reply:
Ah, gotcha, thanks. Could you leave a comment in the code explaining that?